spec/services/course_assignable_subject_service_spec.rb

```diff
@@ -0,0 +1,43 @@
+describe CourseAssignableSubjectService do
+  let(:service) do
+    described_class.new(
+      primary_subject: primary_model,
+      secondary_subject: secondary_model,
+      further_education_subject: further_education_model,
+      modern_language_subject: modern_language_model,
+      modern_languages_parent_subject: modern_languages_parent_subject,
+    )
+  end
+  let(:primary_model) { spy }
+  let(:secondary_model) { spy }
+  let(:further_education_model) { spy }
+  let(:modern_language_model) { spy }
+  let(:modern_languages_parent_subject) { spy }
+
+  it "calls all on primary if level is primary" do
+    course = create(:course, level: "primary")
+    service.execute(course)
```

Comment:
This test, and the ones below, don't check what gets returned. Would that be useful?

Reply:
Yes, they ensure that we're getting data in the correct way, although I should definitely be testing the return values.